src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -60,9 +60,9 @@
      * @return Query object.
      */
     public static QueryAction create(Client client, String sql) throws SqlParseException, SQLFeatureNotSupportedException {
-        sql = sql.replaceAll("\n"," ");
-        String firstWord = sql.substring(0, sql.indexOf(' '));
-        switch (firstWord.toUpperCase()) {
+        sql = sql.replaceAll(System.lineSeparator()," ").trim();
```

Comment:
as remarked by @dai-chen , the user input might not necessarily happen on the same system. Even though this is not ideal, but I suggest doing this to account for both Windows- and Unix-style new lines:
```
sql = sql.replaceAll("\r\n", " ").replaceAll("\n", " ").trim();
```

Reply:
There's a regex that is called "linebreak matcher", I'll use it